compio-runtime/src/lib.rs

```diff
@@ -1,97 +1,28 @@
 //! The runtime of compio.
-//! We don't expose the runtime struct because there could be only one runtime
-//! in each thread.
 //!
 //! ```
-//! let ans = compio_runtime::block_on(async {
+//! let ans = compio_runtime::Runtime::new().unwrap().block_on(async {
 //!     println!("Hello world!");
 //!     42
 //! });
 //! assert_eq!(ans, 42);
 //! ```
 
 #![cfg_attr(feature = "once_cell_try", feature(once_cell_try))]
-#![cfg_attr(feature = "lazy_cell", feature(lazy_cell))]
 #![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
 #![warn(missing_docs)]
 
 mod attacher;
 mod key;
-pub(crate) mod runtime;
+mod runtime;
 
 #[cfg(feature = "event")]
 pub mod event;
 #[cfg(feature = "time")]
 pub mod time;
 
-use std::{cell::RefCell, future::Future, io, mem::ManuallyDrop};
-
 pub use async_task::Task;
 pub use attacher::*;
 use compio_buf::BufResult;
-use compio_driver::{OpCode, ProactorBuilder, RawFd};
 pub(crate) use key::Key;
-// It cannot be replaced by std one because of lacking `get` method.
-use once_cell::unsync::Lazy as LazyCell;
-use runtime::Runtime;
-
-thread_local! {
-    pub(crate) static PROACTOR_BUILDER: RefCell<ProactorBuilder> = RefCell::new(ProactorBuilder::new());
-    // Use ManuallyDrop here to avoid misc bugs on some platforms when
-    // trying to get current thread id after thread local resources dropped.
-    pub(crate) static RUNTIME: LazyCell<ManuallyDrop<Runtime>> = LazyCell::new(|| {
-        ManuallyDrop::new(
-            PROACTOR_BUILDER
-                .with(|builder| Runtime::new(&builder.borrow()))
-                .expect("cannot create compio runtime")
-        )
-    });
-}
-
-/// Config the inner proactor with a [`ProactorBuilder`]. Note that if any
-/// runtime related method is called before, there will be no influence.
-/// The return value indicates whether the builder will be used when
-/// initializing the runtime.
-pub fn config_proactor(new_builder: ProactorBuilder) -> bool {
-    PROACTOR_BUILDER.with(|builder| *builder.borrow_mut() = new_builder);
-    RUNTIME.with(|runtime| LazyCell::get(runtime).is_none())
-}
-
-/// Start a compio runtime and block on the future till it completes.
-pub fn block_on<F: Future>(future: F) -> F::Output {
```

Comment:
Keep `block_on` for ease of use like `spawn`.

Reply:
It won't be easy to use. We cannot call `block_on` recursively.

Instead, I'll provide a `block_on` for EnterGuard.